toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -915,24 +921,23 @@ func createLiveOSIsoImage(buildDir, baseConfigPath string, isoConfig *imagecusto
 //     iso image.
 func extractIsoImageContents(buildDir string, isoImageFile string, isoExpansionFolder string) (err error) {
 
-	mountDir, err := ioutil.TempDir(buildDir, "tmp-iso-mount-")
+	mountDir, err := os.MkdirTemp(buildDir, "tmp-iso-mount-")
 	if err != nil {
 		return fmt.Errorf("failed to create temporary mount folder for iso:\n%w", err)
 	}
 	defer os.RemoveAll(mountDir)
 
-	mountParams := []string{isoImageFile, mountDir}
-	err = shell.ExecuteLive(false, "mount", mountParams...)
+	loopDevice, err := safeloopback.NewLoopback(isoImageFile)
 	if err != nil {
-		return fmt.Errorf("failed to mount iso:\n%w", err)
+		return fmt.Errorf("failed to create loop device for (%s):\n%w", isoImageFile, err)
 	}
-	defer func() {
-		unmountParams := []string{mountDir}
-		cleanupErr := shell.ExecuteLive(false, "umount", unmountParams...)
-		if cleanupErr != nil {
-			err = fmt.Errorf("%w:\nfailed to clean-up (%s): %w", err, mountDir, cleanupErr)
-		}
-	}()
+	defer loopDevice.CleanClose()
```

Comment:
Use `Close` with `defer`. And call `CleanClose` at the bottom of the function, checking the error. There are plenty of examples throughout the codebase.

Reply:
Are we using that pattern because we can't capture the error of a deferred function?

Reply:
pending

Reply:
`Close` requests the resource to be closed but doesn't wait for it. `CleanClose` waits for the close to succeed, with a timeout.

`Close` is typically used in error cases, where you just want to clean everything up but you don't want/need to wait for it. Whereas, `CleanClose` ensures all the bits have fully flushed before moving on.